relbench/tasks/avito.py

```diff
@@ -14,21 +17,125 @@
+        search_stream = db.table_dict["SearchStream"].df
+        timestamp_df = pd.DataFrame({"timestamp": timestamps})
+        df = duckdb.sql(
+            f"""
+            SELECT
+                search_ads.AdID,
+                t.timestamp,
+                COALESCE(SUM(search_ads.isClick), 0) / COALESCE(COUNT(search_ads.SearchID), 1) AS num_click
+            FROM
+                timestamp_df t
+            LEFT JOIN (
+                ads_info
+                    LEFT JOIN
+                search_stream
+                    ON
+                ads_info.AdID == search_stream.AdID
+            ) search_ads
+            ON
+                search_ads.SearchDate > t.timestamp AND
+                search_ads.SearchDate <= t.timestamp + INTERVAL '{self.timedelta} days'
+            GROUP BY
+                t.timestamp,
+                search_ads.AdID
+            HAVING
+                SUM(search_ads.isClick) > 0
+            """
+        ).df()
+
+        return Table(
+            df=df,
+            fkey_col_to_pkey_table={"AdID": "entity_table"},
+            pkey_col=None,
+            time_col="timestamp",
+        )
+
+
+class UserVisitsTask(RelBenchNodeTask):
+    r"""Predict whether each customer will visit more than one ad in the next
+    4 days.
+    """
+
+    name = "user-visits"
+    task_type = TaskType.BINARY_CLASSIFICATION
+    entity_table = "UserInfo"
+    entity_col = "UserID"
+    time_col = "timestamp"
+    target_col = "num_click"
+    timedelta = pd.Timedelta(days=4)
+    metrics = [average_precision, accuracy, f1, roc_auc]
+
+    def make_table(self, db: Database, timestamps: "pd.Series[pd.Timestamp]") -> Table:
+        user_info = db.table_dict["UserInfo"].df
+        visits_stream = db.table_dict["VisitStream"].df
+        timestamp_df = pd.DataFrame({"timestamp": timestamps})
+        df = duckdb.sql(
+            f"""
+            SELECT
+                visit_ads.UserID,
+                t.timestamp,
+                COALESCE(COUNT(DISTINCT visit_ads.AdID), 0) > 1 AS num_click
+            FROM
+                timestamp_df t
+            LEFT JOIN
+            (
+                    user_info
+                LEFT JOIN
+                    visits_stream
+                ON
+                    user_info.UserID == visits_stream.UserID
+            ) visit_ads
+            ON
+                visit_ads.ViewDate > t.timestamp AND
+                visit_ads.ViewDate <= t.timestamp + INTERVAL '{self.timedelta} days'
+            GROUP BY
+                t.timestamp,
+                visit_ads.UserID
+            """
+        ).df()
+
+        return Table(
+            df=df,
+            fkey_col_to_pkey_table={"UserID": "entity_table"},
+            pkey_col=None,
+            time_col="timestamp",
+        )
+
+
 class UserClicksTask(RelBenchNodeTask):
-    r"""Predict the total number of ads each customer will click in the
+    r"""Predict the each customer will click on more than one ads in the
```

Comment:
What's the difference between visit and click?

Reply:
click signal is in SearchStream table, while visit signal is in VisitsStream table.

Reply:
Ok, but is that a completely different task? Are they similar? Do we need both?

Reply:
Yes, they are completely different tasks.